Make cursor position always work

toolness opened the pull request:
This should resolve #80.

It should be noted, though, that this fix is potentially more conservative than it needs to be:
- It abandons all italic/bold styling, which we might actually be able to get away with.
- It sets the font to "monospace", which is whatever the browser defaults to, usually courier new. There are many much nicer free, monospaced fonts out there that we might want to use instead.

toolness committed:
Replaced CodeMirror styling to be extremely conservative.

Font is now monospace, line height doesn't change, the style
is never bold or italic, gutter font is same as editor font.

toolness committed:
Merge branch 'gh-pages' into cursor-position-fix

toolness committed:
make error/help messages non-monospaced.

Pomax commented:
Please do not merge this until I've had been able to reproduce, because I think the solution involving simply setting line-height would be enough to completely fix everything.

toolness committed:
Use Mensch instead of monospace and reduce font size a bit.

toolness committed:
Added Mensch font (forgotten from my last commit).

toolness committed:
Merge branch 'gh-pages' into cursor-position-fix

Conflicts:
	css/errorhelp.css

toolness committed:
Move CSS out of fc/ui/text.js. "normal" now default text size.

Moving the CSS out of fc/ui/text.js also simplifies it a lot, as setting
text size is now just a matter of toggling some classes on the CodeMirror
wrapper, rather than dynamically generating a stylesheet.

This fixes #86.

Also added some code to deal with edge cases if localStorage is full.

Pomax commented:
This also contains various changes that were added to solve the cursor positioning based on the assumption the documentation claim that only monospace is supposed. As it turns out, that documentation is incorrect, as the CodeMirror author stated in the issue file about this that multiple fonts and non-monospaced fonts are irrelevant, with the issue being only related to line-height needing to be consistent. Can we make a new pull request for the changes in 3e41b0840d5cf0b991a03d2d6b44907ec28030d0 applied directly to head without the intermediate commits, and then tie the other commits to styling tickets (if we have them) on webpagemaker?

toolness commented:
Sorry, I should have closed this pull request on friday... Back then I did exactly what you suggested and committed it as ee840492dde8b94f7ac4d5e0a5833e866f809306, but didn't close this pull request, resulting in a very confusing situation.
